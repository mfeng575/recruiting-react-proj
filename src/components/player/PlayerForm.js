import React from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form';
import FieldInput from '../common/FieldInput';

const required = value => (value ? undefined : 'Required')
const minValue = min => value =>
  value && value < min ? `Value must be at least ${min}` : undefined
const maxValue = max => value =>
  value && value > max ? `Value must be at most ${max}` : undefined  
const minValue1 = minValue(1)
const maxValue100 = maxValue(100)

export const PlayerForm = ({ handleSubmit, pristine, reset, submitting, heading, handleSave, handleCancel }) => {
    return (
        <form onSubmit={handleSubmit(handleSave)}>
            <h1>{heading}</h1>

            <Field
                type="text"
                name="firstName"
                label="First Name"
                placeholder=""
                validate={[required]}
                component={FieldInput}
            />

            <Field
                type="text"
                name="lastName"
                label="Last Name"
                placeholder=""
                validate={[required]}
                component={FieldInput}
            />

            <Field
                type="number"
                name="score"
                label="Score"
                placeholder=""
                validate={[required, minValue1, maxValue100]}
                parse={value => isNaN(parseInt(value, 10)) ? null : parseInt(value, 10)}
                component={FieldInput}
            />

            <div>
                <button type="submit" disabled={submitting} className="btn btn-primary"><i className="fa fa-paper-plane-o" aria-hidden="true" /> Submit</button>

                {heading === 'Add' && <button type="button" disabled={pristine || submitting} onClick={reset} className="btn btn-default btn-space">Clear Values</button>}

                <button type="button" className="btn btn-default btn-space" onClick={handleCancel}>Cancel</button>
            </div>
        </form>
    );
};

PlayerForm.propTypes = {
    handleSubmit: PropTypes.func.isRequired,
    pristine: PropTypes.bool.isRequired,
    reset: PropTypes.func.isRequired,
    submitting: PropTypes.bool.isRequired,
    heading: PropTypes.string.isRequired,
    handleSave: PropTypes.func.isRequired,
    handleCancel: PropTypes.func.isRequired
};

export default reduxForm({
    form: 'PlayerForm'
})(PlayerForm);
