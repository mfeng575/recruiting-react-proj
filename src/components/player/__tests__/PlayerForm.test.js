import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import toJson from 'enzyme-to-json';
import {PlayerForm} from '../PlayerForm';



function setup(heading) {
    const props = {
        handleSubmit: jest.fn(),
        pristine: true,
        reset: jest.fn(),
        submitting: false,
        heading: heading,
        handleSave: jest.fn(),
        handleCancel: jest.fn(),
    };

    return shallow(<PlayerForm {...props}/>);
}


describe('PlayerForm.test.js', () => {

    it('renders without crashing', () => {
        const wrapper = setup('Add Player');
        expect(wrapper.length).toEqual(1);
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });


    it('renders form and display "Add Player" in h1', () => {
        const wrapper = setup('Add Player');
        expect(wrapper.find('h1').text()).toEqual('Add Player');
    });


    it('displays "Edit Player" in h1', () => {
        const wrapper = setup('Edit Player');
        expect(wrapper.find('h1').text()).toEqual('Edit Player');
    });

    
});

