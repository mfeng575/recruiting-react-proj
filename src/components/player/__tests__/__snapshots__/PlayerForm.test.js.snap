// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerForm.test.js renders without crashing 1`] = `
<form>
  <h1>
    Add Player
  </h1>
  <Field
    component={[Function]}
    label="First Name"
    name="firstName"
    placeholder=""
    type="text"
    validate={
      Array [
        [Function],
      ]
    }
  />
  <Field
    component={[Function]}
    label="Last Name"
    name="lastName"
    placeholder=""
    type="text"
    validate={
      Array [
        [Function],
      ]
    }
  />
  <Field
    component={[Function]}
    label="Score"
    name="score"
    parse={[Function]}
    placeholder=""
    type="number"
    validate={
      Array [
        [Function],
        [Function],
        [Function],
      ]
    }
  />
  <div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      <i
        aria-hidden="true"
        className="fa fa-paper-plane-o"
      />
       Submit
    </button>
    <button
      className="btn btn-default btn-space"
      onClick={[Function]}
      type="button"
    >
      Cancel
    </button>
  </div>
</form>
`;
